### ISSUE TITLE
very slow and than crashes when many chanels are avalible...

### ISSUE THREAD

**Issue by Evohl** [linuxmint/hypnotix#272]
as soon you try searching for something or sometimes also when try to open channels list the GUI is hanging for around 20- 30 sec then closes. Seems a kind of timeout happens.

[evohl@archlinux ~]$ hypnotix 
[evohl@archlinux ~]$ Free-TV: 988 channels, 46 groups, 0 series, 172 movies
Loading providers...
Free-TV: Getting playlist...
Free-TV: Checking playlist...
mytv: 212692 channels, 0 groups, 5209 series, 0 movies
Free-TV: Loading channels...
mytv: Getting playlist...
mytv: Checking playlist...
mytv: Loading channels...
Filtering 212692 channel names containing the string 'the last of us'...
[evohl@archlinux ~]$ 

Sysinfo:

System:
  Kernel: 6.2.2-arch1-1 arch: x86_64 bits: 64 compiler: gcc v: 12.2.1
    Desktop: Xfce v: 4.18.1 Distro: Arch Linux
Machine:
  Type: Desktop System: ASUS product: N/A v: N/A serial: <superuser required>
  Mobo: ASUSTeK model: ROG STRIX B550-F GAMING WIFI II v: Rev X.0x
    serial: <superuser required> UEFI: American Megatrends v: 2806
    date: 10/27/2022
CPU:
  Info: 12-core model: AMD Ryzen 9 3900X bits: 64 type: MT MCP arch: Zen 2
    rev: 0 cache: L1: 768 KiB L2: 6 MiB L3: 64 MiB
  Speed (MHz): avg: 2533 high: 3800 min/max: 2200/4672 boost: enabled cores:
    1: 3800 2: 3800 3: 2200 4: 2200 5: 3800 6: 2200 7: 2200 8: 2200 9: 2200
    10: 2200 11: 2200 12: 2200 13: 2200 14: 3800 15: 2200 16: 2200 17: 2200
    18: 2200 19: 2200 20: 2200 21: 2200 22: 2200 23: 3800 24: 2200
    bogomips: 182482
  Flags: avx avx2 ht lm nx pae sse sse2 sse3 sse4_1 sse4_2 sse4a ssse3 svm
Graphics:
  Device-1: AMD Navi 21 [Radeon RX 6800/6800 XT / 6900 XT]
    vendor: XFX Speedster MERC 319 driver: amdgpu v: kernel arch: RDNA-2
    bus-ID: 0a:00.0
  Display: x11 server: X.org v: 1.21.1.7 driver: X: loaded: amdgpu
    unloaded: modesetting,radeon dri: radeonsi gpu: amdgpu resolution:
    1: 1920x1080~144Hz 2: 3440x1440~144Hz 3: 1920x1080~144Hz
  API: OpenGL v: 4.6 Mesa 22.3.6 renderer: AMD Radeon RX 6900 XT (navi21
    LLVM 15.0.7 DRM 3.49 6.2.2-arch1-1) direct-render: Yes
Audio:
  Device-1: AMD Navi 21/23 HDMI/DP Audio driver: snd_hda_intel v: kernel
    bus-ID: 0a:00.1
  Device-2: AMD Starship/Matisse HD Audio vendor: ASUSTeK
    driver: snd_hda_intel v: kernel bus-ID: 0c:00.4
  Sound API: ALSA v: k6.2.2-arch1-1 running: yes
  Sound Interface: sndio v: N/A running: no
  Sound Server-1: PulseAudio v: 16.1 running: no
  Sound Server-2: PipeWire v: 0.3.66 running: yes
Network:
  Device-1: MEDIATEK MT7921K Wi-Fi 6E 80MHz driver: N/A bus-ID: 06:00.0
  Device-2: Intel Ethernet I225-V vendor: ASUSTeK driver: igc v: kernel
    port: N/A bus-ID: 07:00.0
  IF: enp7s0 state: up speed: 1000 Mbps duplex: full mac: <filter>
Bluetooth:
  Device-1: MediaTek Wireless_Device type: USB driver: btusb v: 0.8
    bus-ID: 1-5:3
  Report: rfkill ID: hci0 rfk-id: 0 state: up address: see --recommends
Drives:
  Local Storage: total: 3.66 TiB used: 1.4 TiB (38.1%)
  ID-1: /dev/nvme0n1 vendor: Sabrent model: N/A size: 953.87 GiB
    temp: 27.9 C
  ID-2: /dev/nvme1n1 vendor: Crucial model: CT1000P3SSD8 size: 931.51 GiB
    temp: 29.9 C
  ID-3: /dev/sda vendor: SanDisk model: SDSSDH3 2T00 size: 1.82 TiB
Partition:
  ID-1: / size: 937.33 GiB used: 152.27 GiB (16.2%) fs: ext4
    dev: /dev/nvme0n1p2
  ID-2: /boot size: 510 MiB used: 82.5 MiB (16.2%) fs: vfat
    dev: /dev/nvme0n1p1
Swap:
  Alert: No swap data was found.
Sensors:
  System Temperatures: cpu: 35.4 C mobo: N/A gpu: amdgpu temp: 53.0 C
  Fan Speeds (RPM): N/A gpu: amdgpu fan: 0
Info:
  Processes: 399 Uptime: 20m Memory: 31.25 GiB used: 3.15 GiB (10.1%)
  Init: systemd Compilers: gcc: 12.2.1 Packages: 1053 Shell: Bash v: 5.1.16
  inxi: 3.3.25

**Comment by Axel-Erfurt** [comment]
This is probably more of a problem with the playlist.
212692 channels ??

**Comment by Evohl** [comment]
yes the playlist is stupid big but works other devices with poor hardware like VU+ reciver with xstreamity or android IPTV player. Or Kodi with simple IPTV. They are also very slow loading the list but works. I tried some different IPTV players for Linux and Windows, mostly unsuccessful. Your player at least works sometimes :D btw. u are German? Würde die kommunikation vereinfachen :)
